File: source/tb/multiplier_top_tb/gen_vec.c
#include <stdio.h>

int main()
{

	
	int nsamples = 10000;
	unsigned int multiplier, multiplicand;
	unsigned long long result;

	int temp = sizeof(result);
	printf("size of result is %d\n", temp);
	// ***** Open o/p files
	FILE *opa_f = fopen("multiplicand_c.txt", "w");
	FILE *opb_f = fopen("multiplier_c.txt", "w");
	FILE *res_f = fopen("result_c.txt", "w");

	int i;
	for (i=0; i<nsamples; i++) {
		multiplicand = rand();
		multiplier = rand();
		result = (unsigned long long) multiplicand * (unsigned long long) multiplier;
		fprintf(opa_f, "%u\n", multiplicand);
		fprintf(opb_f, "%u\n", multiplier);
		fprintf(res_f, "%llu\n", result);
	}

	fclose(opa_f);
	fclose(opb_f);
	fclose(res_f);

	return 0;
}

File: source/tb/multiplier_top_tb/gen_vec.m

%% Generate random inputs to the multiplier and reference result

% call random function which will generate floating point numbers < 1

nsamples = 10000;

multiplicand = zeros(nsamples,1);
multiplier = zeros(nsamples,1);

for i = 1:nsamples
	multiplicand(i) = rand(1, 'single');
	multiplier(i) = rand(1, 'single');
end

multiplicand = round(multiplicand.*2^32);
multiplier = round(multiplier.*2^32);


% result = multiplicand.*multiplier;

% result = dec2bin(typecast(result, 'uint64'));
%  (typecast(B,'uint32'))

%  gauss_smpls =  gauss_smpls.*(gauss_smpls<=2^11) + (gauss_smpls-2^12).*(gauss_smpls>2^11);

% multiplicand_x = dec2bin(typecast(multiplicand, 'uint32'));
% multiplier_x = dec2bin(typecast(multiplier, 'uint32'));



% dlmwrite('multiplicand_mat.txt', multiplicand, '');
% dlmwrite('multiplier_mat.txt', multiplier, '');


multiplicand_x = dec2bin(multiplicand);
multiplier_x = dec2bin(multiplier);

dlmwrite('multiplicand_mat.txt', multiplicand_x, '');
dlmwrite('multiplier_mat.txt', multiplier_x, '');
% dlmwrite('result.txt', result, '');


multiplicand_c = dlmread('multiplicand_c.txt');
multiplier_c = dlmread('multiplier_c.txt');

% multiplicand_cb = dec2bin(typecast(multiplicand_c, 'uint32'));
% multiplier_cb = dec2bin(typecast(multiplier_c, 'uint32'));

multiplicand_cb = dec2bin(multiplicand_c);
multiplier_cb = dec2bin(multiplier_c);


dlmwrite('multiplicand_cb.txt', multiplicand_cb, '');
dlmwrite('multiplier_cb.txt', multiplier_cb, '');



write_finish = 1